src/errors.ts

```diff
@@ -3,16 +3,39 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { parseError } from "vscode-azureextensionui";
 import { GitErrorCodes } from "./git";
 import { localize } from "./utils/localize";
 
 export class NoWorkspaceError extends Error {
     public message: string = localize('noWorkspaceError', 'You must have a workspace open to perform this operation.');
 }
 
+// use only when using the gitApi
+export function handleGitError(err: unknown): void {
+    if (typeof err === 'object' && err !== null) {
+        const stdout = (<Error & { stdout?: string }>err).stdout;
+        // ignore empty commit errors which will happen if a user initializes a blank folder or have changes in a nested git repo
+        if (stdout && /nothing to commit/.test(stdout)) {
+            return;
+        } else if ('gitErrorCode' in err) {
+            throw new GitError(<GitErrorType>err);
+        }
+
+        // if it's not a GitError, just throw up the original error
+        throw err;
+    } else if (err === null || err === undefined) {
```

Comment:
I'd rather you got rid of these last two cases and just threw up the original error for those, too. Our standard error handling should take care of it